internal/parser/planparserv2/plan_parser_v2.go

```diff
@@ -13,8 +15,25 @@ import (
 	"github.com/milvus-io/milvus/pkg/util/typeutil"
 )
 
+// exprParseKey is used to cache the parse result. Currently only collectionName is used besides expr string, which implies
+// that the same collectionName will have the same schema thus the same parse result. In the future, if there is case that the
+// schema changes without changing the collectionName, we need to change the cache key.
+type exprParseKey struct {
+	collectionName string
```

Comment:
prefer to use `CollectionID` here, which could works with `collection alias` and `collection rename`

Reply:
That is preferable, however, it may involve a lot of modifications, and it may be not usable if the schema is able to change in the future.  

We decided to keep as is per our offline discussion.